File: LeetCode_JAVA/11. Container With Most Water.java
class Solution {
    public int maxArea(int[] height) {
        int left =0;
        int right = height.length-1;
        
        int max=0;
        
        while (left < right) {
            
            int x = right- left;
            
            int y = Math.min(height[left], height[right]);
            max = Math.max(max, x * y);
            
            if (height[left] > height[right]) {
                right--;
            } else {
                left++;
            }
        }
        
        return max;
    }
}

File: Leetcode_JavaScript/11. Container With Most Water.js
/**
 * @param {number[]} height
 * @return {number}
 */
var maxArea = function(height) {
    let left = 0;
    let right = height.length -1;
    let max = 0;
    while(left < right){
        let x = right -left;
        let y = Math.min(height[left], height[right]);
        max = Math.max(max, x*y);
        if(height[left] < height[right]){
            left++;
        }else{
            right--;
        }
    }
    return max;
    
};
